sklearn/ensemble/_hist_gradient_boosting/grower.py

```diff
@@ -264,28 +264,19 @@ def __init__(
             has_missing_values = [has_missing_values] * X_binned.shape[1]
         has_missing_values = np.asarray(has_missing_values, dtype=np.uint8)
 
+        # Shallow validation of monotonic_cst to make TreeGrower easier to
+        # test. A more complete validation is done in _validate_monotonic_cst
+        # at the estimator level and therefore the following should not be
+        # needed when using the public API.
```

Comment:
With this PR, I think all the validation is now at the estimator level? In that case:

```suggestion
        # `monotonic_cst` validation is done in _validate_monotonic_cst
        # at the estimator level and therefore the following should not be
        # needed when using the public API.
```